### ISSUE TITLE
MEMORY_BRAM pass doesn't not consider potential RAMs

### ISSUE THREAD

**Issue by CalebKierum** [YosysHQ/yosys#1711]
I have created the following module in my FPGA using a template provided to me. 

```
module instruction_fetch (
  input wire [31:0] address,
  input wire clk,
  output reg [31:0] instruction
);

  reg [31:0] code_mem [1023:0];

  initial begin
    $readmemb("code.mem", code_mem);
  end
  always @(posedge clk) begin
    instruction <= code_mem[address >> 2];
  end
endmodule

```

It would be ideal for this module (and others like it) to be brams on my iCE40 TinyFPGA BX. However when I build this I get the following log

[build.log](https://github.com/YosysHQ/yosys/files/4228777/build.log)

Here is an excerpt with just the BRAM pass log 

[bram.log](https://github.com/YosysHQ/yosys/files/4228774/bram.log)

I know it is not truncating signals such as code_mem, register_file, and membigtagthing as in simulation those memories are successfully used. It is just this synthesis step.

In summary Yosis does not appear to be checking if these registers could be a bram and I am fairly sure it should.
